### PR TITLE
Add cln channel acceptor

### PR DESCRIPTION
**Pull request by JssDWt**

### CODE REVIEW

File: cln_plugin/cln_messages.go [hunk 1]
```diff
@@ -115,3 +115,24 @@ type LogNotification struct {
...
+	MaxAcceptedHtlcs         uint32 `json:"max_accepted_htlcs"`
+	ChannelFlags             uint64 `json:"channel_flags"`
+}
+
+type OpenChannelResult struct {
+	Result      string  `json:"result"`
+	CloseTo     *string `json:"close_to,omitempty"`
+	MinDepth    *uint32 `json:"mindepth,omitempty"`
+	ReserveMsat *string `json:"reserve,omitempty"`
+}
```
Comment (JssDWt) [inline cln_plugin/cln_messages.go:138]: These can be removed now

File: cln_plugin/cln_plugin.go [hunk 2]
```diff
@@ -382,6 +437,67 @@ func (c *ClnPlugin) handleHtlcAccepted(request *Request) {
 	c.server.Send(idToString(request.Id), &htlc)
 }
 
+func handleSetChannelAcceptScript(request *Request) {
+	// TODO
```
Comment (JssDWt) [inline cln_plugin/cln_plugin.go:441]: TODO

**Review by JssDWt — COMMENTED** [review]
Looks good, I don't really understand the execution of the starlark script, but if that spits out a valid openchannel response a la `{ "result": "continue" }` or `{ "result": "reject" }` this should work.

Some small comments along the way.

**Comment by yaslama** [comment]
> Looks good, I don't really understand the execution of the starlark script, but if that spits out a valid openchannel response a la `{ "result": "continue" }` or `{ "result": "reject" }` this should work.

The script can be for instance:
```(lambda oc: encode({"result":"continue"}) if oc["id"]=="d988690bdf6036ed530aab4baf09cfc81f7fcd853e8d595a16e1a4a818af956222" else encode({"result":"reject", "error_message": "cannot open a channel"}))(decode(openchannel))```

**Pull request merged**